### PR TITLE
[osgearth] Fix errors that occur in the osgearth config file on some platforms

### PR DESCRIPTION
**Pull request by cbrl**
**Describe the pull request**

- #### What does your PR fix?
  Fixes the exported target in osgEarthConfig.cmake. Though it seems to work on Windows, on other platforms (e.g. Linux) the capitalization of the `find_path` argument causes the operation to fail, and the lack of the library definition causes an error upon attempting to set that target's properties.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  Unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

### CODE REVIEW

**Comment by Cheney-W** [comment]
Local testing in progress.

File: ports/osgearth/fix-osgearth-config.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+-set(OSGEARTH_NAMES osgearth)
++set(OSGEARTH_NAMES osgEarth)
+ 
+ find_library(osgearth_LIBRARY
+     NAMES ${OSGEARTH_NAMES}
+@@ -38,6 +38,8 @@ find_package_handle_standard_args(osgearth DEFAULT_MSG
+ 
+ string (REPLACE ";" " " osgearth_LDFLAGS "${osgearth_LDFLAGS}")
+ 
++add_library(osgearth @OSGEARTH_DYNAMIC_OR_STATIC@ IMPORTED)
```
Comment (ras0219-msft) [inline ports/osgearth/fix-osgearth-config.patch:23]: It looks like upstream intends this to be `osgEarth`[1].
```suggestion
+add_library(osgEarth UNKNOWN IMPORTED)
```
[1]: https://github.com/gwaldron/osgearth/blob/master/osgEarthConfig.cmake.in

**Review by ras0219-msft — DISMISSED** [review]
Thanks for the PR! LGTM with the one suggestion applied.

File: ports/osgearth/fix-osgearth-config.patch (hunk 1 above)
Comment (cbrl) [reply ports/osgearth/fix-osgearth-config.patch:23]: Yes, I was unsure whether to keep the target name from the current release, or the newer target name.  That one is technically the correct spelling, so I'll commit the suggestion.

**Comment by Cheney-W** [comment]
versions/o-/osgearth.json#L2
versions/o-/osgearth.json(2,): error : While reading versions for port osgearth from file: version/o-/osgearth.json
File declares version `3.3#1` with SHA: 1198ef439cf8f6851e6b3accbe668c34a46c8ba6
But local port with the same version has a different SHA: 1091743df235c6472d30b157723f82e6d3473cc9
Please update the port's version fields and then run:

    vcpkg x-add-version osgearth
    git add versions
    git commit -m "Update version database"

to add a new version.

**Comment by Cheney-W** [comment]
Please don't change `port-version` twice in one PR, you need to change the value of `port-version` back to 1, and then execute the following command:
```
1. <modify file>
2. git add .
3. git commit -m "xxx"
4. .\vcpkg.exe x-add-version osgearth --overwrite-version
5. git add .
6. git commit --amend --no-edit
7. git push
```

**Comment by cbrl** [comment]
Sorry about that. I amended the modification with the proper port version.

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**